engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/selectModulesScreen/SelectModulesScreen.java

```diff
@@ -403,6 +433,32 @@ private void filterText(String newText) {
         }
     }
 
+
+    private void moduleWiseFilter() {
+        sortedModules.clear();
+        for (ModuleSelectionInfo m : allSortedModules) {
+            if (selectModulesConfig.isLibraryChecked() && StandardModuleExtension.isLibraryModule(moduleManager.getRegistry().getLatestModuleVersion(m.getMetadata().getId()))) {
```

Comment:
I see possible improvements here, for example you could extract `moduleManager.getRegistry().getLatestModuleVersion(m.getMetadata().getId())` to a variable instead of repeating it in each check.
A more elaborate refactoring could be extracting a method like `moduleMatchesFilter(Module module, StandardModuleExtension filter)` but it would require more changes, like extracting the possible filters (asset, gameplay, etc) to a custom enum, and saving a `java.util.Set` of said enum in the config and as a single extension rather than multiple boolean extensions and config fields.
But probably it's not necessary for now since it's quite readable anyway (in my understanding at least).